src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -652,10 +652,12 @@ bool emberAfOperationalCredentialsClusterAddNOCCallback(app::CommandHandler * co
     VerifyOrExit(!failSafeContext.NocCommandHasBeenInvoked(), nonDefaultStatus = Status::ConstraintError);
 
     VerifyOrExit(NOCValue.size() <= 400, nonDefaultStatus = Status::InvalidCommand);
+    VerifyOrExit(ICACValue.HasValue() && ICACValue.Value().size() <= 400, nonDefaultStatus = Status::InvalidCommand);
```

Comment:
Actually, this will fail if  `ICAC.HasValue()` is false. Thinking more about it you need:

```
if (ICACValue.HasValue())
{
    VerifyOrExit(ICACValue.Value().size() <= 400, nonDefaultStatus = Status::InvalidCommand);
}
```

And same in UpdateNOC

Reply:
The version with the if() seems more easy to understand for me. Otherwise you are subtly playing out Demorgan's theorem in your mind since VerifyOrExit is already reverse logic ("exit if condition false")

Reply:
Sure, I am OK with it either way.